### PR TITLE
Add integration tests for Dashboard and API key generation

### PR DESCRIPTION
**Pull request by kachkaev**
In scope
- Dashboard page accessibility for guests and logged in users
- Dashboard page key elements and links
- API key page key elements
- API key page key generation
- API key page key re-generation
- API key page key modal edge cases
- Reusable API-based login flow that skips UI (to be used in tests)
- Playwright 1.24

Out of scope
- Type generation from the dashboard

### CODE REVIEW

File: patches/playwright-core+1.24.0.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+--- a/node_modules/playwright-core/lib/server/fetch.js
++++ b/node_modules/playwright-core/lib/server/fetch.js
+@@ -187,7 +187,9 @@ class APIRequestContext extends _instrumentation.SdkObject {
+       const cookie = parseCookie(header);
+       if (!cookie) continue; // https://datatracker.ietf.org/doc/html/rfc6265#section-5.2.3
+ 
+-      if (!cookie.domain) cookie.domain = url.hostname;else (0, _utils.assert)(cookie.domain.startsWith('.'));
++      // This hack allows setting cookies in site/tests/shared/nav.ts → login
++      // To be removed in https://app.asana.com/0/1202542409311090/1202652337622563/f
++      if (!cookie.domain) cookie.domain = url.hostname;else cookie.domain = cookie.domain.replace(/^./, '');
```
Comment (kachkaev) [inline patches/playwright-core+1.24.0.patch:12]: I faced a rather odd Playwright bug that was not allowing me to set cookies. This patched helped avoid this issue. TODO (later): investigate this code and submit upstream issue if needed: https://github.com/microsoft/playwright/blame/99d1ad5a88c3e89360829eee92dbaa98d75beaa4/packages/playwright-core/src/server/fetch.ts#L210-L213

Tracked in https://app.asana.com/0/1202542409311090/1202652337622563/f (internal)

File: site/package.json [hunk 2]
```diff
@@ -91,7 +91,7 @@
   },
   "devDependencies": {
     "@babel/runtime": "^7.17.9",
-    "@playwright/test": "1.23.3",
+    "@playwright/test": "^1.24.0",
```
Comment (kachkaev) [inline site/package.json:94]: I updated Playwright in hope to avoid the patch, but it was still needed. We got improved clipboard support in Chrome by doing this too. Release notes: https://github.com/microsoft/playwright/releases

**Review by teenoh — APPROVED** [review]
Thanks @kachkaev!
